server/mscalendar/availability.go

```diff
@@ -27,6 +29,9 @@ const (
 	upcomingEventNotificationWindow = (StatusSyncJobInterval * 11) / 10 // 110% of the interval
 	logTruncateMsg                  = "We've truncated the logs due to too many messages"
 	logTruncateLimit                = 5
+
+	// defaultConcurrency is the default number of workers to span for calendar providers that doesn't allow batch requests
+	defaultConcurrency = 4
```

Comment:
Why was this value picked? What are the tradeoffs?

Reply:
I have zero insights into customers environments, but that seemed a reasonable number to have to achieve proper concurrency/parallelism if running server has at least 4c/t. Using a higher number can give better results but also diminishing returns if the server can't handle all at once, and I don't want to hammer a lot of goros to the scheduler to just let them wait for each other. If this where a CLI tool or something I would use `runtime.NumCPU()` but I don't want the plugin to use all available resources every 5 minutes, even if most of the time it's going to be waiting for responses. What do you think? Is there any other plugin doing something similar?

Reply:
>  don't want the plugin to use all available resources every 5 minutes, even if most of the time it's going to be waiting for responses.

I don't think that's what is happening here. As you pointed out, the workers will mostly wait for an HTTP response. During that time the go runtime is smart enough to pass control of the CPU to other processes.

 >  Is there any other plugin doing something similar?
 
 Not on that scale. E.g. the GitHub plugin does concurrent API requests, but only four.
 
 Do you know if the MS calendar API punishes a lot of requests coming from the same IP? I.e. can we use an aggressive number here?

Reply:
> Do you know if the MS calendar API punishes a lot of requests coming from the same IP? I.e. can we use an aggressive number here?

Yes, that was the other reason I forgot about yesterday. Google APIs have per project and per user quotas. Though the documentation says that its for write requests, I don't know if we can trigger some limit if we do a lot of reads too.

Reply:
Yes, but we are building the gcal plugin on top of the mscalendar, by using different remotes when retrieving data, but maintaining the plugin UI/UX.

Reply:
In that case, I would suggest moving the concurrency variable out of the function (maybe into `mscalendar`?) and configure it per backend.

Reply:
I can do that as well, though the mscalendar allows for batch requests so it's never going to reach this piece of code.